src/coreclr/nativeaot/Runtime/riscv64/UniversalTransition.S

```diff
@@ -0,0 +1,178 @@
+        fsd  fa5, 0x38(sp)         // Save f5
+        fsd  fa6, 0x40(sp)         // Save f6
+        fsd  fa7, 0x48(sp)         // Save f7
+
+        // Space for return buffer data (0x40 bytes)
+
+        // Save argument registers
+        sd  a0, 0xb0(sp)           // Save a0
+        sd  a1, 0xb8(sp)           // Save a1
+        sd  a2, 0xc0(sp)           // Save a2
+        sd  a3, 0xc8(sp)           // Save a3
+        sd  a4, 0xd0(sp)           // Save a4
+        sd  a5, 0xd8(sp)           // Save a5
+        sd  a6, 0xe0(sp)           // Save a6
+        sd  a7, 0xe8(sp)           // Save a7
+
+        // Optionally prepare the values to trash saved argument registers
+    #ifdef TRASH_SAVED_ARGUMENT_REGISTERS
+        PREPARE_EXTERNAL_VAR RhpFpTrashValues, a1
+
+        fld  fa0, 0(a1)            // Load fa0 from a1
+        fld  fa1, 0x08(a1)         // Load fa1 from a1
+        fld  fa2, 0x10(a1)         // Load fa2 from a1
+        fld  fa3, 0x18(a1)         // Load fa3 from a1
+        fld  fa4, 0x20(a1)         // Load fa4 from a1
+        fld  fa5, 0x28(a1)         // Load fa5 from a1
+        fld  fa6, 0x30(a1)         // Load fa6 from a1
+        fld  fa7, 0x38(a1)         // Load fa7 from a1
+
+        PREPARE_EXTERNAL_VAR RhpIntegerTrashValues, a1
+
+        ld  a2, 0x10(a1)           // Load a2 from a1
+        ld  a3, 0x18(a1)           // Load a3 from a1
+        ld  a4, 0x20(a1)           // Load a4 from a1
+        ld  a5, 0x28(a1)           // Load a5 from a1
+        ld  a6, 0x30(a1)           // Load a6 from a1
+        ld  a7, 0x38(a1)           // Load a7 from a1
+    #endif // TRASH_SAVED_ARGUMENT_REGISTERS
+
+        addi a0, sp, DISTANCE_FROM_CHILDSP_TO_RETURN_BLOCK  // First parameter to target function
+        mv a1, t1                                         // Second parameter to target function
+        jalr t0, t1, 0                                    // Jump to the function in t1
+
+        // Restore the result address from t2
+        mv t2, a0                                       // Move result to t2
+
+        // Restore Floating Point registers
+        fld f0, 0x100(sp)         // Restore f0
+        fld f1, 0x108(sp)         // Restore f1
+        fld f2, 0x110(sp)         // Restore f2
+        fld f3, 0x118(sp)         // Restore f3
+        fld f4, 0x120(sp)         // Restore f4
+        fld f5, 0x128(sp)         // Restore f5
+        fld f6, 0x130(sp)         // Restore f6
+        fld f7, 0x138(sp)         // Restore f7
+
+        // Restore argument registers
+        ld a0, 0x140(sp)         // Restore a0
+        ld a1, 0x148(sp)         // Restore a1
+        ld a2, 0x150(sp)         // Restore a2
+        ld a3, 0x158(sp)         // Restore a3
+        ld a4, 0x160(sp)         // Restore a4
+        ld a5, 0x168(sp)         // Restore a5
+        ld a6, 0x170(sp)         // Restore a6
+        ld a7, 0x178(sp)         // Restore a7
+        ld tp, 0x180(sp)         // Restore tp
```

Comment:
I don't see tp (thread pointer) saved above. It's not an argument register, does it need to be handled here at all?

Reply:
It was translated from LoongArch64 assembly, but seeing that it was removed in https://github.com/dotnet/runtime/commit/2f37b48292361f7ba95f35dad70e6763d67c7811, lets clean this up
```suggestion
```